File: TeachersRating.API/Entities/Worker.cs
namespace TeachersRating.API.Entities;

public class Worker : BaseEntity
{
    public string FullName { get; set; } = string.Empty;
    public string Position { get; set; } = string.Empty;
    public string PersonalPageUrl { get; set; } = string.Empty;

    public Guid PhotoId { get; set; }
    public Photo? Photo { get; set; } = null;

    public uint NumberOfLikes { get; set; }
    public uint NumberOfDislikes { get; set; }

    public ICollection<Department> Departments { get; set; } = new List<Department>();
}

File: NulpScrapper/models/Worker.ts
import type { Department } from "./Department.js";

export interface Worker {
    imageUrl: string;
    fullName: string;
    position: string;
    personalPageHref: string;
    department: Department;
}

export function createWorker(imageUrl: string, fullName: string, position: string, personalPageHref: string, department: Department): Worker {
    return { imageUrl, fullName, position, personalPageHref, department } as Worker;
}